src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -168,27 +183,41 @@ public sealed class CultureAwareComparer : StringComparer
         private readonly CompareInfo _compareInfo; // Do not rename (binary serialization)
         private readonly bool _ignoreCase; // Do not rename (binary serialization)
 
+        [OptionalField]
+        private CompareOptions _options;
+
         internal CultureAwareComparer(CultureInfo culture, bool ignoreCase)
         {
             _compareInfo = culture.CompareInfo;
             _ignoreCase = ignoreCase;
+            _options = _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        }
+
+        internal CultureAwareComparer(CultureInfo culture, CompareOptions compareOptions)
+        {
+            _compareInfo = culture.CompareInfo;
+            _options = compareOptions;
         }
 
-        private CompareOptions Options => _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        [OnDeserialized]
+        private void OnDeserializedMethod(StreamingContext context)
+        {
+            _options |= _ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
+        }
```

Comment:
I know I have suggested this way but I was looking same time at issue on the full framework. after investigating it  I found it is better to have CultureAwareComparer implement ISerializable something like 


```C#
        public CultureAwareComparer(SerializationInfo info, StreamingContext context)
        {
            _compareInfo = (CompareInfo) info.GetValue("_compareInfo", typeof(CompareInfo));
            bool ignoreCase = info.GetBoolean("_ignoreCase");

            var obj = info.GetValueNoThrow("_options", typeof(CompareOptions));
            if (obj != null)
                _options = (CompareOptions) obj;

            // fix up the _options value in case we are getting old serialized object not having _options
            _options |= ignoreCase ? CompareOptions.IgnoreCase : CompareOptions.None;
        }

        public void GetObjectData(SerializationInfo info, StreamingContext context)
        {
            info.AddValue("_compareInfo", _compareInfo);
            info.AddValue("_options", _options);
            info.AddValue("_ignoreCase", _(options & CompareOptions.IgnoreCase) != 0);
        }

```

note that you can fully remove _ignoreCase now too.